### ISSUE TITLE
[Feature]: Add task perplexity mode to optimize PPL evaluation

### ISSUE THREAD

**Issue by efsotr** [vllm-project/vllm#16324]
### 🚀 The feature, motivation and pitch

When testing 2/3/4-bit quantized LLaMA 2 models (7B/13B/70B) on a single A40 GPU using `lm-evaluation-harness`, I ran into issues with `generate` mode for perplexity (PPL) tasks:

1. PPL evaluation typically uses `generate` mode with `prompt_log_probs=0` and `max_tokens=1`.
2. The current implementation reuses the response log-prob calculation for prompt log-probs. However, the number of new response tokens (≈ number of requests) is much smaller than the total number of prompt tokens. Since response log-probs are computed assuming short outputs, the code uses a memory-heavy approach that easily leads to OOM when applied to long prompts — requiring reduced `gpu_memory_utilization`. 
see https://github.com/vllm-project/vllm/blob/24f6b9a71397539a3d02c801963220b0e9a2aef9/vllm/model_executor/layers/sampler.py#L268-L284
3. Lowering `gpu_memory_utilization` introduces another issue: it keeps KV cache around, causing errors. (This can be mitigated by lowering `max_model_len`, but it's a workaround. However, setting `max_model_len`  too low may lead to a large number of requests, which in turn could cause OOM issues.) 
see https://github.com/vllm-project/vllm/blob/24f6b9a71397539a3d02c801963220b0e9a2aef9/vllm/worker/worker.py#L543-L550

### Alternatives

Add a new `task perplexity` mode that returns the same output as `prompt_log_probs=0`, but:
- Optimizes the logits computation path specifically for PPL tasks
- Avoids using KV cache 

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.
